lib/spack/llnl/util/filesystem.py

```diff
@@ -1683,3 +1682,18 @@ def prefixes(path):
         pass
 
     return paths
+
+
+def md5sum(file):
```

Comment:
I would remove this function and replace its usage with the builtin `filecmp` module

Reply:
@adamjstewart It's not clear to me how that would be possible. What I'm testing is that two files that don't exist at the same time on the filesystem have the same content. When a user does:
```console
spack config update packages
```
what should happen is that:
- Files in the new format gets written in `packages.yaml`
- Files in the old format are moved to `packages.yaml.bkp` and kept as a backup copy

In the tests I am checking that the md5sum of `packages.yaml` before this operation matches that of `packages.yaml.bkp` after the operation.